odl/trafos/wavelet.py

```diff
@@ -652,21 +982,28 @@ def _call(self, x):
             The length of the array depends on the size of input image to
             be transformed and on the chosen wavelet basis.
         """
-        if x.space.ndim == 1:
-            coeff_list = pywt.wavedec(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 2:
-            coeff_list = pywt.wavedec2(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 3:
-            coeff_dict = wavelet_decomposition3d(x, self.wbasis, self.mode,
-                                                 self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_dict, self.size_list)
-
+        if self._axes is None:
+            if x.space.ndim == 1:
+                coeff_list = pywt.wavedec(x, self.wbasis, self.mode,
+                                          self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+                return self.range.element(coeff_arr)
+
+            if x.space.ndim == 2:
+                coeff_list = pywt.wavedec2(x, self.wbasis, self.mode,
+                                           self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+                return self.range.element(coeff_arr)
+
+            if x.space.ndim == 3:
+                coeff_list = wavelet_decomposition3d(x, self.wbasis, self.mode,
+                                                     self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+
+                return self.range.element(coeff_arr)
+        else:
+            coeff_dict = pywt.dwtn(x, self.wbasis, self.mode, self._axes)
```

Comment:
Then these two functionalities are quite different. The first one is (as I understand it) the wavelet decomposition that stores the minimum amount of coefficients necessary to uniquely reconstruct the original signal. The second one seems to take inner products with either the scaling function or the wavelet in the provided axes, including the usual size reduction. This seems to be closer to the notion of "Continuous wavelet transform".

Comparing sizes, one gets
```python
>>> 32 ** 2
1024
>>> sum(np.size(val) for val in using_wavedec2)
1267
>>> sum(np.size(val) for val in using_dwtn.values())
1600
```

Do the PyWavelets guys have a document lying around that explains what exactly is computed? Something like [this](http://www.fftw.org/doc/The-1d-Discrete-Fourier-Transform-_0028DFT_0029.html#The-1d-Discrete-Fourier-Transform-_0028DFT_0029)?